### ISSUE TITLE
Database migrations for SQL databases to preserve data

### ISSUE THREAD

**Issue by saydus** [eveningkid/denodb#178]
It would be awesome to have migrations as in [sequelize](https://github.com/sequelize/sequelize) or [typeORM](https://github.com/typeorm/typeorm/blob/master/docs/migrations.md).

**Comment by FinlayDaG33k** [comment]
Yes, migrations would be desired as it makes it easier to setup a new database (eg. when you develop a new device), makes database structures more predictable and lowers the chance something goes horribly wrong.  
Additionally, it makes it easier for use with CI/CD where direct access to the database may not always be possible.
